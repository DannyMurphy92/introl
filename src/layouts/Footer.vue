<template>
  <div class="footer">
    <div class="site-trunk footer_container">
      <ContactInfo class="footer__contact" />
      <div class="footer__legal">
        <span>Copyright &copy; Introl {{ currentYear }}. All rights reserved.</span>
      </div>
    </div>
  </div>
</template>
<script lang="ts">
import Vue from 'vue';
import ContactInfo from '@/components/ContactInfo.vue';

export default Vue.extend({
  name: 'Footer',
  components: {
    ContactInfo,
  },
  computed: {
    currentYear() {
      return new Date(Date.now()).getFullYear();
    },
  },
});
</script>

<style lang="scss">
.footer {
  background: $color-grey;
  padding: $footer-vertical-padding 0;
  width: 100%;
  height: $footer-height;
}
</style>
