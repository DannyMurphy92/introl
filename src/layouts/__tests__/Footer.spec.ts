import Footer from '../Footer.vue';
import { shallowMount, Wrapper } from '@vue/test-utils';

describe('Footer', () => {
  let wrapper: Wrapper<any>;
  const mockDate = new Date('1999-01-02T00:00:00Z');

  const mountComponent = () => {
    wrapper = shallowMount(Footer);
  };

  beforeEach(() => {
    jest.spyOn(global.Date, 'now').mockImplementation(() => mockDate.valueOf());
    mountComponent();
  });

  it('Then renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('THEN should render the legal text correctly', () => {
    expect(wrapper.find('.footer__legal').text()).toBe('Copyright Â© Introl 1999. All rights reserved.');
  });
});
