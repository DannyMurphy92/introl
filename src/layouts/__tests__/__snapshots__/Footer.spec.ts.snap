// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Then renders correctly 1`] = `
<div class="footer">
  <div class="site-trunk footer_container">
    <contactinfo-stub class="footer__contact"></contactinfo-stub>
    <div class="footer__leagal"><span>Copyright Â© Introl 1999. All rights reserved.</span></div>
  </div>
</div>
`;
